### PR TITLE
Feature: Run Kafka streams app with Pulsar

### PR DESCRIPTION
**Pull request by dlg99**
### Motivation

Provide an easy way to run Kafka Streams applications with Pulsar.
Depends on https://github.com/apache/pulsar-adapters/pull/9

### Modifications

Upgraded to Kafka client 2.7
Fixed Consumer and Producer logic that was not enough for KafkaStreams to run.
Added/shaded noop AdminClient to prevent Stream from trying to use kafka's one.

### Verifying this change

Added example and description on how to run it.

Integration tests do not pass (even without the changes), did not try to add the new one.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): yes

### Documentation

  - Does this pull request introduce a new feature? kind of
  - If yes, how is the feature documented? Readme, example, comments.

### CODE REVIEW

**Comment by dlg99** [comment]
The easiest way to review is to look at the last two commits, previous commits are from pre-req https://github.com/apache/pulsar-adapters/pull/10

**Comment by sijie** [comment]
@BewareMyPower Can you review this PR?

File: examples/kafka-streams/pom.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-api</artifactId>
+      <version>2.6.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-core</artifactId>
+      <version>2.6.1</version>
+    </dependency-->
```
Comment (BewareMyPower) [inline examples/kafka-streams/pom.xml:61]: Should we remove it?

File: pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java [hunk 2]
```diff
@@ -366,7 +372,15 @@ private RecordMetadata getRecordMetadata(String topic, TypedMessageBuilder<byte[
...
         TopicPartition tp = new TopicPartition(topic, partition);
         TypedMessageBuilderImpl<byte[]> mb = (TypedMessageBuilderImpl<byte[]>) msgBuilder;
-        return new RecordMetadata(tp, offset, 0L, mb.getPublishTime(), 0L, mb.hasKey() ? mb.getKey().length() : 0, size);
+
+        long publishTime = 0L;
+        try {
+            // there is no hasPublishTime() currently
+            publishTime = mb.getPublishTime();
+        } catch (IllegalStateException ise) {
+            logger.debug("could not get publish time");
```
Comment (BewareMyPower) [inline pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java:381]: Why is the log level debug? I think `sendAsync` will set the publish time and `getPublishTime()` failure is an unexpected case so here the log level should be error.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM as soon as @BewareMyPower comments are addressed.

We should also create a minimal documentation about how to run Kafka Streams.
The example is great, but we must write on Pulsar website that this feature is present and it works.

Please create an issue on Pulsar main repository for the docs.

File: pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java (hunk 2 above)
Comment (dlg99) [reply pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java:381]: all messages that passed through LIneSplit example had this problem. 
Error logging was too noisy.
I didn't have time to investigate where it is expected to be filled in / why it was blank.

Comment (eolivelli) [reply pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java:381]: @dlg99 good point. +1

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**